contracts/marketplace/Marketplace.sol

```diff
@@ -44,18 +45,18 @@ contract Marketplace {
     token.manualTransfer(seller, msg.sender, amountToBuy);
 
     //Transfer ETH to the seller
-    msg.sender.transfer(msg.value);
+    sellP.ownerAddress.transfer(msg.value);
   }
 
   //TODO: Save sell positions to database
 
   function endSellPostion(DebtToken token) public {
-    sellPosition memory sellP = sellPositions[msg.sender][token.name()];
+    sellPosition memory sellP = sellPositions[msg.sender][address(token)];
 
     //Check if sell position exists
     require(sellP.exists);
```

Comment:
Maybe preferable to simply check if field 'amount' is non-zero and remove field 'exists'?